### PR TITLE
Rewrite Gatsby API from gatsby folder to TS

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Changes proposed in this pull request:
- Rewrite Gatsby API from gatsby folder to TS

**Related issue(s)**
https://github.com/kyma-project/website/issues/287

### CODE REVIEW

File: tslint.json [hunk 1]
```diff
@@ -31,6 +31,6 @@
   },
   "jsRules": {},
   "linterOptions": {
-    "exclude": ["node_modules/**/*", "tools/**/*"]
+    "exclude": ["*/node_modules/**/*"]
   }
 }
```
[the remark this answers is not in the document]
Comment (aerfio) [reply tslint.json]: even though it doesn't lint `./gatsby`, just for the future

[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply tslint.json]: I think that `String` object may be used in code 😄

File: gatsby/config/plugins.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      name: "content",
+      path: `${__dirname}/../../content/`,
+    },
+  },
+  {
+    resolve: "gatsby-plugin-copy-files",
+    options: {
+      source: `${__dirname}/../../content`,
+      destination: "/assets",
+      extensions: ["jpeg", "jpg", "gif", "png", "svg", "json", "yaml", "yml"],
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply gatsby/config/plugins.ts]: We use it in one place, so extracting is unnecessary.

File: gatsby/index.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+"use strict";
+
+require("source-map-support").install();
+require("ts-node").register({
+  compilerOptions: {
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply gatsby/index.js]: Thanks!

File: package.json [hunk 4]
```diff
@@ -132,8 +134,7 @@
         "git add"
       ],
       "src/**/*.{ts,tsx}": [
-        "tslint -c tslint.json --project tsconfig.json --format verbose --fix",
-        "prettier --config .prettierrc --write",
+        "npm run lint-fix",
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply package.json]: `npm run lint-fix` is short command of tslint and pretiier, so I think it works like with options.

File: gatsby/config/rss-feed.ts [hunk 5]
```diff
@@ -1,4 +1,33 @@
 import { siteMetadata } from "./sitemetadata";
+import { SiteMetadata } from "./types";
+
+interface Edge {
+  node: {
+    excerpt: string;
+    html: string;
+    fields: {
+      slug: string;
```
Comment (aerfio) [inline gatsby/config/rss-feed.ts]: `slug: string?`, at least from what you said to me

**Pull request merged**